File: go/src/leetcode/hot100/LeetCode206.go
package hot100

func reverseList(head *ListNode) *ListNode {
	if head == nil {
		return nil
	}
	var pre, next = head, head.Next
	pre.Next = nil
	for next != nil {
		temp := next.Next
		next.Next = pre
		pre = next
		next = temp
	}
	return pre
}

File: java/src/leetcode/problems/LeetCode206.java
package leetcode.problems;

import common.*;

/**
 * @author wujingxinit@outlook.com
 * @date 2024/4/25 14:34
 */
public class LeetCode206 {

    public ListNode reverseList(ListNode head) {
        ListNode pre = null;
        ListNode cur = head;
        while (cur != null) {
            ListNode next = cur.next;
            cur.next = pre;
            pre = cur;
            cur = next;
        }
        return pre;
    }
}
